federation/pkg/kubefed/init/init.go

```diff
@@ -0,0 +1,502 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// TODO(madhusdancs):
+// 1. Add a dry-run support.
+// 2. Make all the API object names customizable.
```

Comment:
ok, you can provide the user support :-)  I still think that the value to the user of allowing users this level of customization would not warrant the additional support complexity.